tests/src/JIT/opt/InstructionCombining/DivToMul.cs

```diff
@@ -0,0 +1,175 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+// JIT is able to replace "x / 2" with "x * 0.5" where 2 is a power of two float
+// Make sure this optimization doesn't change the results
+
+public class Program
+{
+    private static int resultCode = 100;
+
+    public static int Main(string[] args)
+    {
+        // Some corner cases
+        var floatValues = new List<float>(
+            new[] { 0, 1, MathF.PI, MathF.E, float.MinValue, float.MinValue, int.MaxValue, int.MinValue, float.NegativeInfinity, float.PositiveInfinity, float.NaN, (float) double.MaxValue, (float) double.MinValue });
```

Comment:
Is negative zero relevant?

Reply:
I've just added more cases. These values then passed to the tests additionally with `-`. So the tests actually have `0.0 / c` and `-0.0 / c` cases (as well as `x / 0.0` and `x / -0.0`)